src/platform_impl/apple/appkit/app_state.rs

```diff
@@ -47,44 +46,21 @@ pub(super) struct AppState {
     // as such should be careful to not add fields that, in turn, strongly reference those.
 }
 
-declare_class!(
-    #[derive(Debug)]
-    pub(super) struct ApplicationDelegate;
-
-    unsafe impl ClassType for ApplicationDelegate {
-        type Super = NSObject;
-        type Mutability = mutability::MainThreadOnly;
-        const NAME: &'static str = "WinitApplicationDelegate";
-    }
-
-    impl DeclaredClass for ApplicationDelegate {
-        type Ivars = AppState;
-    }
-
-    unsafe impl NSObjectProtocol for ApplicationDelegate {}
-
-    unsafe impl NSApplicationDelegate for ApplicationDelegate {
-        #[method(applicationDidFinishLaunching:)]
-        fn app_did_finish_launching(&self, notification: &NSNotification) {
-            self.did_finish_launching(notification)
-        }
-
-        #[method(applicationWillTerminate:)]
-        fn app_will_terminate(&self, notification: &NSNotification) {
-            self.will_terminate(notification)
-        }
-    }
-);
+// TODO(madsmtm): Use `MainThreadBound` once that is possible in `static`s.
+thread_local! {
+    static GLOBAL: OnceCell<Rc<AppState>> = const { OnceCell::new() };
```

Comment:
What's the reason to `Rc` thread local global? I'd assume the main reason is that the user creates it in `Rc` and shares the pointer with winit, thus it's correctly ref-counted.

And you're using global here just for simplicity?

Reply:
> `Rc`

I haven't really decided how I want to pass the state around yet, so I just went with something that worked, and matched the semantics of what we were doing before (`Retained<ApplicationDelegate>` is basically the same as `Rc<AppState>`).

> thread local

That was to avoid a bit of `unsafe` until I've released a version of `objc2-foundation` with a `const` `MainThreadBound::new`. But I've worked around it with a helper for now instead.

> global

I _think_ that the only piece of state that has to _actually_ be global (in Winit) is the `ApplicationHandler`, since we have to access that from our overwritten `WinitApplication` when (hackily) dispatching device events. But if I find a different way to implement that, then I might be able to remove this final piece of global state in Winit's AppKit backend.